tcp/pcb.ml

```diff
@@ -465,6 +468,10 @@ struct
       Lwt.return_unit
     | None ->
       Tx.send_rst t id ~sequence ~ack_number ~syn ~fin
+    | exception ex ->
+      PCB_Log.err (fun f -> f "uncaught exception looking up handler for TCP port %d:@ %s"
+                      id.WIRE.local_port (Printexc.to_string ex));
+      Lwt.return_unit
```

Comment:
This is quibbling, but this `~listeners` isn't necessarily from the direct stack.  It's an internal hashtable carried in the record t ( https://github.com/mirage/mirage-tcpip/blob/master/tcp/pcb.ml#L84 ).  I think the relevant guards are here, in `new_server_connection`: https://github.com/mirage/mirage-tcpip/blob/master/tcp/pcb.ml#L369